stdlib/public/SDK/Foundation/Foundation.swift

```diff
@@ -429,6 +432,9 @@ extension NSNumber
 }
 
 extension NSNumber : _HasCustomAnyHashableRepresentation {
+  // Must be @nonobjc to prevent infinite recursion trying to bridge
+  // AnyHashable to NSObject.
```

Comment:
Probably. It was hitting `NSString` specifically, but @nonobjc doesn't hurt for the others either.